src/core/Akka/Serialization/NewtonSoftJsonSerializer.cs

```diff
@@ -143,29 +143,38 @@ public NewtonSoftJsonSerializer(ExtendedActorSystem system, Config config)
         public NewtonSoftJsonSerializer(ExtendedActorSystem system, NewtonSoftJsonSerializerSettings settings)
             : base(system)
         {
+            var settingsSetup = system.Settings.Setup.Get<NewtonSoftJsonSerializerSetup>()
+                .GetOrElse(NewtonSoftJsonSerializerSetup.Create(() => new JsonSerializerSettings
+                {
+                    PreserveReferencesHandling = settings.PreserveObjectReferences 
+                        ? PreserveReferencesHandling.Objects 
+                        : PreserveReferencesHandling.None,
+                    NullValueHandling = NullValueHandling.Ignore,
+                    DefaultValueHandling = DefaultValueHandling.Ignore,
+                    MissingMemberHandling = MissingMemberHandling.Ignore,
+                    ConstructorHandling = ConstructorHandling.AllowNonPublicDefaultConstructor,
+                    TypeNameHandling = settings.EncodeTypeNames
+                        ? TypeNameHandling.All 
+                        : TypeNameHandling.None,
+                    ContractResolver = new AkkaContractResolver()
+                }));
+
+            Settings = settingsSetup.CreateSettings();
```

Comment:
Oh, yes, good point. I should apply the HOCON changes first, then apply the Setup changes. In that sense, this approach is not good, I should do the ASP.NET-ish approach of passing an `Action` with the setting as a parameter instead, letting the user change the setting that way instead of asking them to provide a fresh setting object.